### PR TITLE
PS-7769 - Fix use-after-return error in audit_log_exclude_accounts_validate

### PR DESCRIPTION
**Pull request by ldonoso**
*Problem:*

`st_mysql_value::val_str` might return a pointer to `buf` which after
the function called is deleted. Therefore the value in `save`, after
reuturnin from the function, is invalid.

In this particular case, the error is not manifesting as `val_str`
returns memory allocated with `thd_strmake` and it does not use `buf`.

*Solution:*

Allocate memory with `thd_strmake` so the memory in `save` is not local.

### CODE REVIEW

File: plugin/audit_log/audit_log.c [hunk 1]
```diff
@@ -1593,24 +1593,31 @@ static MYSQL_THDVAR_STR(query_stack,
                         PLUGIN_VAR_NOSYSVAR | PLUGIN_VAR_NOCMDOPT,
                         "Query stack.", NULL, NULL, "");
 
+static const char *val_strmake(MYSQL_THD              thd,
+                               struct st_mysql_value *mysql_val)
+{
+  char        buf[80];
```
Comment (oleksandr-kachan) [inline plugin/audit_log/audit_log.c]: Lets probably use STRING_BUFFER_USUAL_SIZE here

File: plugin/audit_log/audit_log.c [hunk 2]
```diff
@@ -1593,24 +1593,31 @@ static MYSQL_THDVAR_STR(query_stack,
...
+  int         len= sizeof(buf);
+  const char *val= mysql_val->val_str(mysql_val, buf, &len);
+
+  if (val)
```
Comment (oleksandr-kachan) [inline plugin/audit_log/audit_log.c]: "val != NULL" ?

**Review by oleksandr-kachan — COMMENTED** [review]
Please take a look at comments and add some description for the PR. Lets also run MTR tests (including ASAN enabled) for the plugin.

**Comment by ldonoso** [comment]
> Please take a look at comments and add some description for the PR. Lets also run MTR tests (including ASAN enabled) for the plugin.

https://ps57.cd.percona.com/job/percona-server-5.7-asan-param/51/

**Review by oleksandr-kachan — APPROVED** [review]

**Pull request merged**